### DISCUSSION TITLE
Drawn MIDI curves are steppy on hardware synths

### DISCUSSION THREAD

**Discussion by MegaloMIDIac** [SynthstromAudible/DelugeFirmware#603] [Ideas]
### Please describe the problem:

If you draw a MIDI curve in automation view with the pads on a MIDI CC, it will sound steppy on hardware synths. If you record MIDI CC from the external synth all is fine. 

### What is the expected behavior?

Smooth transitions 

### What hardware did you reproduce it with?

7SEG

### What firmware did you reproduce it with?

nightly

**Comment by m-m-adams** [comment]
Can you enable interpolation on the external synth? 

Interpolating all midi to a high resolution on the deluge would significantly increase the amount of CC data sent which 
1) impacts note on/off timing since it's a serial protocol
2) requires more cpu time in the deluge

The recording works fine because your synth is sending CCs at a higher rate than the deluge automation currently generates them (1 per step at current zoom level). We can play with that frequency a little but without interpolating externally it will always be a bit steppy

**Comment by MegaloMIDIac** [comment]
I have tested this behaviour with the Microfreak, Ambika and Blofeld. I don't think they can interpolate incoming MIDI CC.
You can hear this especially if you use the filter cutoff CC. Unfortunately it's not a little bit steppy but sounds like a sidechain. 
Would an Deluge internal CC interpolation not be possible?

**Comment by seangoodvibes** [comment]
This is with interpolation on in automation view right?

How are you inputting the MIDI automation lanes on the deluge? 

If you are changing one step at a time, it will be stepped regardless of whether interpolation is on or off. 

But if you enter a long press (press two pads to connect them) it will be interpolated when interpolation is on and stepped when interpolation is off.

**Comment by MegaloMIDIac** [comment]
I tried both, with and without interpolation in automation view. 
Also the long press will get you the steppy result unfortunately.
I also recorded to Ableton. Made a 1 bar loop on the Deluge with a CC curve from 0 to 128.
In Ableton you can see, it does not create a straight line or curve, but only steps.

**Comment by seangoodvibes** [comment]
Thanks for reporting this @MegaloMIDIac 

I just confirmed the same behaviour sending MIDI to my dirtywave m8.

I think solving this is out of scope for the first release as it's a bigger change that will require some thought.

Cheers

Sean

**Comment by m-m-adams** [comment]
Yeah the automation view sets a value for each pad at the current zoom level. Properly smoothing to only change cc value by 1 at a time will require deeper changes and still sound steppy on many synths
